File: 자료구조/10828.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});
let input = [];
let cnt = 0;
let N = 0;
let stack = [];

const isEmpty = () => {
  if (stack.length === 0) return true;
  else return false;
};

rl.on('line', (line) => {
  if (cnt === 0) N = parseInt(line.trim());
  else if (cnt <= N + 1) {
    input.push(line.trim().split(' '));
  }
  cnt++;
  if (cnt === N + 1) rl.close();
});
rl.on('close', () => {
  for (let i = 0; i < N; i++) {
    if (input[i][0] === 'push') stack.push(input[i][1]);
    else if (input[i][0] === 'pop') {
      if (isEmpty()) console.log(-1);
      else console.log(stack.pop());
    } else if (input[i][0] === 'size') console.log(stack.length);
    else if (input[i][0] === 'empty') {
      if (isEmpty()) console.log(1);
      else console.log(0);
    } else if (input[i][0] === 'top') {
      if (isEmpty()) console.log(-1);
      else console.log(stack.at(-1));
    }
  }

  process.exit();
});

File: 자료구조withPy/10828.py
import sys
N=int(input())

stack=[]

def isEmpty(tmp):
    if len(tmp)==0:return True
    else: return False

for _ in range(N):
    order=sys.stdin.readline().split()
    
    if order[0]=="push":stack.append(int(order[1]))
    elif order[0]=="pop":
        if(isEmpty(stack)): print(-1)
        else: print(stack.pop())
    elif order[0]=="size":print(len(stack))
    elif order[0]=="empty":
        if(isEmpty(stack)): print(1)
        else: print(0)
    elif order[0]=="top":
        if(isEmpty(stack)): print(-1)
        else:print(stack[-1])
    
    


        